akka-actor/src/main/scala/akka/util/Unsafe.java

```diff
@@ -23,8 +27,66 @@
             }
             if (found == null) throw new IllegalStateException("Can't find instance of sun.misc.Unsafe");
             else instance = found;
+            stringValueFieldOffset = instance.objectFieldOffset(String.class.getDeclaredField("value"));
+
+            // See Oracle section 1.5.3 at:
+            // https://docs.oracle.com/javase/8/docs/technotes/guides/versioning/spec/versioning2.html
+            final int[] version = Arrays.
+                stream(System.getProperty("java.specification.version").split("\\.")).
+                mapToInt(Integer::parseInt).
+                toArray();
+            javaVersion = version[0] == 1 ? version[1] : version[0];
         } catch (Throwable t) {
             throw new ExceptionInInitializerError(t);
         }
     }
+
+    public static void copyUSAsciiStrToBytes(String str, byte[] bytes) {
```

Comment:
s/Str/String/

Reply:
Do you mean `String Str` instead of `String str`? since it is a Java class I used the camel case convention?

Reply:
In the method name "StringToBytes" :)

Reply:
Can rename from `copyUSAsciiStrToBytes` to `copyStringToBytes` with the caveat that for Java 8 non-US ASCII are going to be truncated of their char representation needs more than 1 char.

Hence the preservation of `USAscii` in the name

Reply:
I meant `copyUSAsciiStringToBytes`, but it doesn't matter. Thanks for great contribution.